src/evict.c

```diff
@@ -402,7 +402,12 @@ int getMaxmemoryState(size_t *total, size_t *logical, size_t *tofree, float *lev
         if (level) *level = 0;
         return C_OK;
     }
-    if (mem_reported <= server.maxmemory && !level) return C_OK;
+
+    if (server.maxmemory_reserved_scale) {
+        if (mem_reported <= server.maxmemory_available && !level) return C_OK;
+    } else if (mem_reported <= server.maxmemory && !level) {
+        return C_OK;
+    }
```

Comment:
I think on a high level we should just replace `maxmemory` with `maxmemory_available` in this function, including L401. We can avoid changing any logic in this function. `maxmemory_available` is a drop-in replacement for `maxmemory`, in other words.